### PR TITLE
feat: models mapping

### LINKED ISSUE #21: Use the mapping from "models" to create the data from the shapefiles

### PR DESCRIPTION
**Pull request by cl-shaharshavit-rnd**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                       |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

Closes #21

### CODE REVIEW

File: src/layerCreator/models/metadataMapper.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { IPropSHPMapping, LayerMetadata, ShapeFileType } from '@map-colonies/mc-model-types';
+import { injectable } from 'tsyringe';
+import { FeatureCollection, GeoJSON } from 'geojson';
+import _ from 'lodash';
```
Comment (asafMasa) [inline src/layerCreator/models/metadataMapper.ts:4]: import { get } from ....

File: src/layerCreator/models/trigger.ts [hunk 2]
```diff
@@ -64,7 +43,7 @@ export class Trigger {
       const productGeoJson = await this.shpParser.parse(productShp, productDbf);
       const metaDataGeoJson = await this.shpParser.parse(metadataShp, metadataDbf);
       //TODO: add error handling for parsing failure (due to invalid file or file still being copied)
-      const metadata = this.parseToMetadata(productGeoJson, metaDataGeoJson, files);
+      const metadata = this.metadataMapper.map(productGeoJson, metaDataGeoJson, filesGeoJson);
```
Comment (asafMasa) [inline src/layerCreator/models/trigger.ts:46]: try catch on the parsing (maping) section?

File: tests/unit/layerCreator/models/metadataMapper.spec.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  beforeAll(() => {
+    loadTestData();
+  });
+
+  beforeEach(() => {
+    metadataMapper = new MetadataMapper();
+  });
+
+  describe('map', () => {
+    it('metadata is map according to model', () => {
```
Comment (asafMasa) [inline tests/unit/layerCreator/models/metadataMapper.spec.ts:24]: do you want to add test for incorrect mapping vs actual file?

**Review by asafMasa — CHANGES_REQUESTED** [review]
Great Job!

File: src/layerCreator/models/metadataMapper.ts (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply src/layerCreator/models/metadataMapper.ts:4]: changed

File: src/layerCreator/models/trigger.ts (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply src/layerCreator/models/trigger.ts:46]: this method cant throw any catchable error so there is no need for a catch

File: tests/unit/layerCreator/models/metadataMapper.spec.ts (hunk 3 above)
Comment (cl-shaharshavit-rnd) [reply tests/unit/layerCreator/models/metadataMapper.spec.ts:24]: what do you mean?
the mapping cant be incorrect as its from the models.
the mapping also don't validate the models so missing data in the source will just result in empty values

**Review by asafMasa — APPROVED** [review]

**Pull request merged**